lib/memory-store.js

```diff
@@ -6,41 +6,60 @@ function calculateNextResetTime(windowMs) {
   return d;
 }
```

Comment:
`calculateNextResetTime` should also take the precision into account. It might need to look through the buckets too (?) We basically want it to give out the next time that the current user could make a request.

Reply:
There is iterator which takes precision into account and moves onto next bucket when this time expires:
```js
  const bucketTime = windowMs / precision;
  const bucketResetInterval = setInterval(this.resetBucket, bucketTime);
```

`NextResetTime` is just pure `windowMs`, since it is sent via HTTP headers to inform the client when will server forget his requests. That happens after all buckets are cleared.

I might have wrong understanding of this header though. If you are not convinced I can change it to also take `precision` into account. :)

Reply:
After reading [https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html#rfc.section.3.3](https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html#rfc.section.3.3) I see that I've understood it wrongly.  Thought it is used in header with time when user will be 100% safe to send new request.